src/main/kotlin/bsu/cc/parser/XlsxParser.kt

```diff
@@ -0,0 +1,82 @@
+package bsu.cc.parser
+
+import org.apache.poi.ss.usermodel.Cell
+import org.apache.poi.ss.usermodel.CellType
+import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import java.io.File
+import java.io.FileInputStream
+import java.lang.IllegalArgumentException
+import java.lang.reflect.Type
+
+
+fun <T> fileToDataClasses(
+        fileName: String,
+        dataProducer: (rowMap: Map<Int, Cell>) -> T,
+        excludedRowIndices: Set<Int>? = null
+): Sequence<T> {
+    return fileToRows(fileName).asSequence().withIndex().filter { (index, _) ->
+        excludedRowIndices == null || !excludedRowIndices.contains(index)
+    }.map { (_, row) ->
+        rowToDataClass(row, dataProducer)
+    }
+}
+
+fun fileToRows(fileName: String): MutableIterator<Row> {
```

Comment:
I was wondering if there may be times where an external class may want to operate on the rows directly rather than the cell. Currently it could just be private though.